src/tests/suite/dfuse_test.c

```diff
@@ -277,21 +306,191 @@ do_ioctl(void **state)
+	times[0]         = now;
+	times[1].tv_sec  = now.tv_sec - 10;
+	times[1].tv_nsec = 20;
+	rc               = futimens(fd, times);
+	assert_return_code(fd, errno);
+	rc = fstat(fd, &stbuf);
+	assert_return_code(rc, errno);
+	assert_int_equal(stbuf.st_mtim.tv_sec, times[1].tv_sec);
+	assert_int_equal(stbuf.st_mtim.tv_nsec, times[1].tv_nsec);
+	prev_ts.tv_sec  = stbuf.st_mtim.tv_sec;
+	prev_ts.tv_nsec = stbuf.st_mtim.tv_nsec;
+
+	/* Repeat the write test again */
+	rc = write(fd, input_buf, sizeof(input_buf));
+	assert_return_code(rc, errno);
+	rc = fstat(fd, &stbuf);
+	assert_return_code(rc, errno);
+	assert_true(timespec_gt(stbuf.st_mtim, prev_ts));
+
+	rc = close(fd);
+	assert_return_code(rc, errno);
+
+	rc = unlinkat(root, "my_file", 0);
+	assert_return_code(rc, errno);
+
+	rc = close(root);
+	assert_return_code(rc, errno);
+}
+
+static int
+run_specified_tests(const char *tests, int *sub_tests, int sub_tests_size)
 {
-	int                     index = 0;
-	int                     opt;
-	struct option           long_options[] = {{"test-dir", required_argument, NULL, 'M'},
-						  {NULL, 0, NULL, 0} };
+	int nr_failed = 0;
+
+	if (strlen(tests) == 0)
+		tests = all_tests;
+
+	while (*tests != '\0') {
+		switch (*tests) {
+		case 'i':
+			printf("\n\n=================");
+			printf("dfuse IO tests");
+			printf("=====================");
+			const struct CMUnitTest io_tests[] = {
+			    cmocka_unit_test(do_openat),
+			    cmocka_unit_test(do_ioctl),
+			};
+			nr_failed = cmocka_run_group_tests(io_tests, NULL, NULL);
+			break;
+		case 's':
+			printf("\n\n=================");
+			printf("dfuse streaming tests");
+			printf("=====================");
+			const struct CMUnitTest stream_tests[] = {
+			    cmocka_unit_test(do_stream),
+			};
+			nr_failed = cmocka_run_group_tests(stream_tests, NULL, NULL);
+			break;
+
+		case 'm':
+			printf("\n\n=================");
+			printf("dfuse metadata tests");
+			printf("=====================");
+			const struct CMUnitTest metadata_tests[] = {
+			    cmocka_unit_test(do_mtime),
+			};
+			nr_failed = cmocka_run_group_tests(metadata_tests, NULL, NULL);
+			break;
+
+		default:
+			assert_true(0);
+		}
 
-	const struct CMUnitTest tests[] = {
-	    cmocka_unit_test(do_openat),
-	    cmocka_unit_test(do_ioctl),
-	};
+		tests++;
+	}
+
+	return nr_failed;
+}
 
-	while ((opt = getopt_long(argc, argv, "M:", long_options, &index)) != -1) {
+int
+main(int argc, char **argv)
+{
+	char                 tests[64];
+	int                  ntests          = 0;
+	int                  nr_failed       = 0;
+	int                  nr_total_failed = 0;
+	int                  opt = 0, index = 0;
+
+	static struct option long_options[] = {{"test-dir", required_argument, NULL, 'M'},
+					       {"all", no_argument, NULL, 'a'},
+					       {"io", no_argument, NULL, 'i'},
+					       {"stream", no_argument, NULL, 's'},
+					       {"metadata", no_argument, NULL, 'm'},
+					       {NULL, 0, NULL, 0}};
```

Comment:
```suggestion
					       {NULL, 0, NULL, 0} };
```

Reply:
Shouldn't we properly resolve this? FWIW, this style doesn't have that problem:
```
static struct option long_options[] = {
    {...}
}
```

Reply:
Yes we should.  I'm thinking I'll push a PR to remove checkpatch warnings that conflict with our clang-format settings.